src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -295,5 +295,47 @@ public static void CopyTo<T>(this T[] array, Span<T> destination)
         {
             new ReadOnlySpan<T>(array).CopyTo(destination);
         }
+
+        /// <summary>
+        /// Determines whether two sequences overlap.
+        /// </summary>
+        public static unsafe bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second)
+        {
+            if (second.IsEmpty)
+                return false;
+
+            ref T firstRef = ref first.DangerousGetPinnableReference();
+            ref T secondRef = ref second.DangerousGetPinnableReference();
+
+            IntPtr firstByteCount = Unsafe.ByteOffset(ref firstRef, ref Unsafe.Add(ref firstRef, first.Length));
+            IntPtr secondByteCount = Unsafe.ByteOffset(ref secondRef, ref Unsafe.Add(ref secondRef, second.Length));
+
+            IntPtr diff = Unsafe.ByteOffset(ref firstRef, ref secondRef);
+
+            return (sizeof(IntPtr) == sizeof(int))
+                ? (uint)diff < (uint)firstByteCount || (uint)diff > (uint)-(int)secondByteCount
+                : (ulong)diff < (ulong)firstByteCount || (ulong)diff > (ulong)-(long)secondByteCount;
+        }
+
+        /// <summary>
+        /// Determines whether two sequences overlap.
+        /// </summary>
+        public static unsafe bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> second)
+        {
+            if (second.IsEmpty)
+                return false;
+
+            ref T firstRef = ref first.DangerousGetPinnableReference();
+            ref T secondRef = ref second.DangerousGetPinnableReference();
+
```

Comment:
Replace the two `ByteOffset`+`Add` with `first.Length * (nuint)Unsafe.SizeOf<T>()`, probably need to rewrite the code to something like:
```csharp
IntPtr diff = Unsafe.ByteOffset(ref firstRef, ref secondRef);
if (sizeof(IntPtr) == sizeof(int))
{
   uint firstByteCount = (uint)first.Length * Unsafe.SizeOf<T>();
   uint secondByteCount = (uint)second.Length * Unsafe.SizeOf<T>();
   return ((uint)diff < firstByteCount) || ((uint)diff > (uint)-(int)secondByteCount);
}
else
{
   ulong firstByteCount = (ulong)first.Length * Unsafe.SizeOf<T>();
   ulong secondByteCount = (ulong)second.Length * Unsafe.SizeOf<T>();
   return ((ulong)diff < firstByteCount) || ((ulong)diff > (ulong)-(long)secondByteCount);
}
```